app/src/main/java/com/kamron/pogoiv/ShareHandlerActivity.java

```diff
@@ -0,0 +1,37 @@
+package com.kamron.pogoiv;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.v4.content.LocalBroadcastManager;
+import android.os.Bundle;
+
+import java.io.IOException;
+
+public class ShareHandlerActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_handler);
+        finish();
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null && type.startsWith("image/")) {
+            Uri imageUri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
+            try {
+                Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), imageUri);
+                Intent newintent = new Intent("process-bitmap");
+                newintent.putExtra("bitmap", bitmap);
+                LocalBroadcastManager.getInstance(ShareHandlerActivity.this).sendBroadcast(newintent);
```

Comment:
@harkin, thanks for the heads up. I initially didn't want to serialize the bitmap in this intent because that's  lot of data to serialize and unserialize, but wanted to keep the receiving Intenthandler generic (so that it can handle files, etc) and this seemed nice. But it might be better to just pass the Uri and let the final intent get the bitmap.

Any idea if the actual bitmap associated with the Uri will still be kept around if I finish handling the this intent from Google Now? From, Google Now's perspective, once this function returns, it could consider the Intent handled. Why would it still be holding on to the bitmap instead of throwing it away?

I think I have to extract the bitmap here. I'm considering a wait and watch approach on this one. Will fix it if anyone reports the issue.